src/phpDocumentor/ConfigurationFactory/PhpDocumentor2.php

```diff
@@ -0,0 +1,181 @@
+     *
+     * @param \SimpleXMLElement $phpDocumentor
+     *
+     * @return array
+     */
+    public function convert(\SimpleXMLElement $phpDocumentor)
+    {
+        $this->validate($phpDocumentor);
+
+        $extensions         = [];
+        $markers            = [];
+        $visibility         = 'public';
+        $defaultPackageName = 'Default';
+        $template           = 'clean';
+        $ignoreHidden       = true;
+        $ignoreSymlinks     = true;
+
+        if (isset($phpDocumentor->parser)) {
+            $extensions = $this->buildExtensionsPart($phpDocumentor->parser);
+            $markers    = $this->buildMarkersPart($phpDocumentor->parser);
+
+            $visibility         = ((string) $phpDocumentor->parser->visibility) ?: $visibility;
+            $defaultPackageName = ((string) $phpDocumentor->parser->{'default-package-name'}) ?: $defaultPackageName;
+            $template           = ((string) $phpDocumentor->transformations->template->attributes()->name) ?: $template;
+
+            if (isset($phpDocumentor->parser->files)) {
+                if (isset($phpDocumentor->parser->files->{'ignore-hidden'})) {
+                    $ignoreHidden = filter_var(
+                        $phpDocumentor->parser->files->{'ignore-hidden'},
+                        FILTER_VALIDATE_BOOLEAN
+                    );
+                }
+
+                if (isset($phpDocumentor->parser->files->{'ignore-symlinks'})) {
+                    $ignoreSymlinks = filter_var(
+                        $phpDocumentor->parser->files->{'ignore-symlinks'},
+                        FILTER_VALIDATE_BOOLEAN
+                    );
+                }
+            }
+        }
+
+        $outputDirectory = ((string) $phpDocumentor->parser->target) ?: 'file://build/docs';
+        $directories     = ((array) $phpDocumentor->parser->files->directory) ?: ['src'];
+
+        $sourcePaths = [];
+        foreach ($directories as $directory) {
+            $sourcePaths[] = (string) (new Dsn($directory))->getPath();
+        }
+
+        $phpdoc2Array = [
+            'phpdocumentor' => [
+                'paths'     => [
+                    'output' => (string) (new Dsn($outputDirectory))->getPath(),
+                    'cache'  => '/tmp/phpdoc-doc-cache',
+                ],
+                'versions'  => [
+                    '1.0.0' => [
+                        'folder' => '',
+                        'api'    => [
+                            'format'               => 'php',
+                            'source'               => [
+                                'dsn'   => 'file://.',
+                                'paths' => $sourcePaths,
+                            ],
+                            'ignore'               => [
+                                'hidden'   => $ignoreHidden,
+                                'symlinks' => $ignoreSymlinks,
+                                'paths'    => ['src/ServiceDefinitions.php'],
```

Comment:
This is not extracted from the XML file while it is present; also: the value `src/ServiceDefinitions.php` was an example and should not be used. The paths should contain a list of files and folders that may be ignored by the parser